binding.cc

```diff
@@ -1406,46 +1407,34 @@ rocksdb_native_read(js_env_t *env, js_callback_info_t *info) {
   return NULL;
 }
 
-static js_value_t *
-rocksdb_native_write_init(js_env_t *env, js_callback_info_t *info) {
+static js_arraybuffer_t
+rocksdb_native_write_init(js_env_t *env) {
   int err;
 
-  js_value_t *handle;
+  js_arraybuffer_t handle;
 
   rocksdb_native_write_batch_t *req;
-  err = js_create_arraybuffer(env, sizeof(rocksdb_native_write_batch_t), (void **) &req, &handle);
+  err = js_create_arraybuffer(env, req, handle);
   assert(err == 0);
 
   req->env = env;
-  req->handle.data = (void *) req;
+  req->handle.data = req;
 
   return handle;
 }
 
-static js_value_t *
-rocksdb_native_write_buffer(js_env_t *env, js_callback_info_t *info) {
+static js_arraybuffer_t
+rocksdb_native_write_buffer(
+  js_env_t *env,
+  js_arraybuffer_span_of_t<rocksdb_native_write_batch_t, 1> req,
+  uint32_t capacity
+) {
   int err;
 
-  size_t argc = 2;
-  js_value_t *argv[2];
-
-  err = js_get_callback_info(env, info, &argc, argv, NULL, NULL);
-  assert(err == 0);
-
-  assert(argc == 2);
-
-  rocksdb_native_write_batch_t *req;
-  err = js_get_arraybuffer_info(env, argv[0], (void **) &req, NULL);
-  assert(err == 0);
-
-  uint32_t capacity;
-  err = js_get_value_uint32(env, argv[1], &capacity);
-  assert(err == 0);
-
-  js_value_t *handle;
+  js_arraybuffer_t handle;
 
   rocksdb_write_t *writes;
-  err = js_create_arraybuffer(env, capacity * sizeof(rocksdb_write_t), (void **) &writes, &handle);
+  err = js_create_arraybuffer(env, capacity * sizeof(rocksdb_write_t), writes, handle);
```

Comment:
```suggestion
  err = js_create_arraybuffer(env, capacity, writes, handle);
```

`sizeof()` must not be used when using the templated functions as the function already knows the size of `T`: https://github.com/holepunchto/libjstl/blob/34a7894383a5174353530852416dbb5738a7fafe/include/jstl.h#L3816